### DISCUSSION TITLE
Members don't seem to have a banner attribute?

### DISCUSSION THREAD

**Discussion by glass-ships** [Rapptz/discord.py#7864] [Q&A] [answered]
`version: discord.py==2.0.0a4074+g95deb553`

It could just be my implementation so I thought I'd ask here first before creating an issue or something. 
I made a simple `banner` command that returns an embed with a user's banner: 
```python 
@commands.command(name="banner", aliases=['b'])
async def banner(self, ctx, *, user: discord.Member=None):
    """
    Returns a user's Discord banner
    if a user is mentioned, or self if not.
    """
    if not user:
        u = ctx.author
    else:
        u = user
    userBannerUrl = u.banner.url
    embed = discord.Embed(description=f"{u.mention}'s banner")
    embed.set_image(url=f"{userBannerUrl}")
    await ctx.send(embed=embed)
    await asyncio.sleep(3.0)
    await ctx.message.delete()
```
The avatar equivalent works perfectly, but when calling `[p]banner` I keep getting the error: 
`Command raised an exception: AttributeError: 'NoneType' object has no attribute 'url'`

Has anyone else run into this / found a workaround?

**Comment by JDJGInc** [comment, marked as answer]
You need to fetch it, cause discord limitation.

**Reply by glass-ships** [reply]
Ahh, fetch the member or the banner asset?

**Reply by JDJGInc** [reply]
You need to use fetch_member to get the banner asset, it's not given by default.

**Reply by glass-ships** [reply]
Aha I see! Thanks so much, I'll update my code :)

**Reply by glass-ships** [reply]
Thanks again @JDJGInc that did the trick! 

I should add for clarification purposes that, as the [docs](https://discordpy.readthedocs.io/en/master/api.html?highlight=fetch%20member#discord.User.banner) state - it's `Client.fetch_user()` or `self.bot.fetch_user()` in the case of a cog command. 

Works like a charm! 

The updated banner command, in case anyone will find the code useful:
```python
    @commands.command(name="banner", aliases=['b'])
    async def banner(self, ctx, *, user: discord.Member=None):
        """
        Returns a user's Discord banner
        if a user is mentioned, or self if not.
        """
        if not user:
            u = ctx.author
        else:
            u = user
        mem = await self.bot.fetch_user(u.id)
        userBannerUrl = mem.banner

        embed = discord.Embed(description=f"{mem.mention}'s banner")
        embed.set_image(url=f"{userBannerUrl}")
        await ctx.send(embed=embed)
        await asyncio.sleep(3.0)
        await ctx.message.delete()
```

**Comment by rperson1508** [comment]
Fyi you could just use `user = user or ctx.author`
